pkg/stream/sources.go

```diff
@@ -0,0 +1,219 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package stream
+
+import (
+	"context"
+
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+// Just creates an observable with a single item.
+func Just[T any](item T) Observable[T] {
+	return FuncObservable[T](
+		func(ctx context.Context, next func(T), complete func(error)) {
+			go func() {
+				if err := ctx.Err(); err != nil {
+					complete(err)
+				} else {
+					next(item)
+					complete(nil)
+				}
+			}()
+		})
+}
+
+// Stuck creates an observable that never emits anything and
+// just waits for the context to be cancelled.
+// Mainly meant for testing.
+func Stuck[T any]() Observable[T] {
+	return FuncObservable[T](
+		func(ctx context.Context, next func(T), complete func(error)) {
+			go func() {
+				<-ctx.Done()
+				complete(ctx.Err())
+			}()
+		})
+}
+
+// Error creates an observable that fails immediately with given error.
+func Error[T any](err error) Observable[T] {
+	return FuncObservable[T](
+		func(ctx context.Context, next func(T), complete func(error)) {
+			go complete(err)
+		})
+}
+
+// Empty creates an empty observable that completes immediately.
+func Empty[T any]() Observable[T] {
+	return Error[T](nil)
+}
+
+// FromSlice converts a slice into an Observable.
+func FromSlice[T any](items []T) Observable[T] {
+	// Emit items in chunks to reduce overhead of mutex in ctx.Err().
+	const chunkSize = 64
```

Comment:
This seems a nice random number :slightly_smiling_face:

Reply:
Yeah quite random. Also premature optimization as I'm expecting `FromSlice` to just be used in tests for most parts and it's not performance critical...